Packs/Iboss/Integrations/Iboss/Iboss.py

```diff
@@ -0,0 +1,1329 @@
+            raise (e)
+
+    return _impl
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    This Client implements API calls, and does not contain any XSOAR logic.
+    Should only do requests and return data.
+    It inherits from BaseClient defined in CommonServer Python.
+    Most calls use _http_request() that handles proxy, SSL verification, etc.
+    For this  implementation, no special attributes defined
+    """
+
+    def __init__(self, username, password, account_settings_id, proxy: bool, verify: bool):
+        super().__init__(base_url="cloud.iboss.com", proxy=proxy, verify=verify)
+        self.username = username
+        self.password = password
+        self.account_settings_id = account_settings_id
+        self._init_cache_params()
+
+    def _init_cache_params(self):
+        context = get_integration_context()
+
+        # self.cloud_token = context.get("cloud_token")
+        # self.xsrf_token = context.get("xsrf_token")
+        self.primary_gateway = context.get("primary_gateway")
+        self.gateway_xsrf_token = context.get("gateway_xsrf_token")
+        self.expiration = int(context.get("expiration", -1))
+
+        if self.primary_gateway:
+            self._base_url = f"https://{self.primary_gateway}"
+        if self.gateway_xsrf_token:
+            self._headers = {"Cookie": self.gateway_xsrf_token}
+
+    def auth(self):
+        demisto.debug("Re-authenticating")
+
+        token = self._get_cloud_token()
+        xsrf_token, jsessionid, expiration = self._get_cloud_settings_tokens(token)
+        primary_gateway = self._get_primary_gateway(token, jsessionid, xsrf_token)
+        gateway_xsrf_token = self._get_swg_xsrf_token(primary_gateway, token)
+
+        context = get_integration_context()
+        context.update({
+            'cloud_token': token,
+            'xsrf_token': token,
+            'jsessionid': jsessionid,
+            'primary_gateway': primary_gateway,
+            'gateway_xsrf_token': gateway_xsrf_token,
+            'expiration': str(int(parse(expiration).timestamp()))
+        })
+
+        self.cloud_token = context.get("cloud_token")
+        self.xsrf_token = context.get("xsrf_token")
+        self.primary_gateway = context.get("primary_gateway")
+        self.gateway_xsrf_token = context.get("gateway_xsrf_token")
+        self.expiration = int(context.get("expiration"))
+
+        self._base_url = f"https://{self.primary_gateway}"
+        self._headers = {"Cookie": self.gateway_xsrf_token}
+
+        set_integration_context(context)
+        return "ok"
+
+    def _get_cloud_token(self):
+        auth = (self.username, self.password)
+        params = {'ignoreAuthModule': 'true'}
+        result = self._http_request(
+            full_url='https://accounts.iboss.com/ibossauth/web/tokens',
+            method="get",
+            timeout=60,
+            auth=auth,
+            params=params
+        )
+        return result["token"]
```

Comment:
should handle the case 'token' is not a key in the results, or make sure that case is impossible.

Reply:
System should always return token if auth successful (if not, wrapper method throws UnAuth error) but added ValueError just in case